### ISSUE TITLE
frame-mAP or video-mAP?

### ISSUE THREAD

**Issue by yiheo** [kkahatapitiya/Coarse-Fine-Networks#6]
Hi. It's me again.
In your paper and code, mAP was used for performance comparison.
1. Is it frame-mAP? or video-mAP?
2. Do you have any plans to add another one?

Thanks.

**Comment by yiheo** [comment]
1. In `log_coarse_fine_train.txt`, there are several mAP values. Is the last mAP value is actual evaluation?

2. In the paper, mAP was 25.10 with Coarse-Fine model. However, I cannot get that much high value. As your log file, mAP is near 22-23%. How can I get proper result?

**Comment by kkahatapitiya** [comment]
The intended here in temporal activity detection, which means making predictions per-frame. So, the given mAP is for frame-wise predictions.

> Do you have any plans to add another one?

What do you mean?

**Comment by yiheo** [comment]
Thanks for reply. I means, do you have plans to add code about video-mAP evaluation? or Can you guide me how to edit the code for video-mAP values?

**Comment by kkahatapitiya** [comment]
In that case, the task would become recognition (instead of detection). If you still want to do it, 

For labels (one-hot): take max across the temporal axis to get video level labels
For logits: correct thing would be to replace the final Linear layers to make video-level predictions (after average pooling over both spatio-temporal axes). Implementation should already be there, but you have to retrain, at least the last layers. Easy thing would be to consider max over temporal axis and use them as video-level predictions
